src/documents/models.py

```diff
@@ -314,6 +397,83 @@ def thumbnail_file(self):
     def thumbnail_url(self):
         return reverse("fetch", kwargs={"kind": "thumb", "pk": self.pk})
 
+    def set_filename(self, filename):
+        if os.path.isfile(self.filename_to_path(filename)):
+            self.filename = filename
+
+
+def delete_empty_directory(directory):
+    if len(os.listdir(directory)) == 0:
+        try:
+            os.rmdir(directory)
+        except os.error:
```

Comment:
I appreciate the TOCTOU check, but in general in Python it is more common to ask for forgiveness than for permission. I suggest to remove the `len()` check altogether and simply try to remove the directory.

Reply:
done